### PR TITLE
Extend user deletion webui

### PR DESCRIPTION
**Pull request by krauselukas**
- Aligns the behavior of the user `delete!` and `mark_as_spammer!` methods
- Add possibility to leave a admin note through the webui
- Add a link for user deletion to the user profile page (for admins only)

image
image

### CODE REVIEW

**Comment by krauselukas** [comment]
[Introduce User deletion UI (M)](https://trello.com/c/Y7Iv5NXM/2332-introduce-user-deletion-ui-m)

File: src/api/app/components/delete_confirmation_dialog_component.html.haml [hunk 1]
```diff
@@ -5,8 +5,10 @@
         %h5.wrap-text.modal-title{ id: "#{modal_id}-label" }= modal_title
       .modal-body
         %p.wrap-text.confirmation-text= confirmation_text
-
         = form_tag(action, method: method, remote: remote) do
+          - if user_deletion
```
[the remark this answers is not in the document]
Comment (krauselukas) [reply src/api/app/components/delete_confirmation_dialog_component.html.haml]: @hellcp-work please have another look, I'm using a more generic approach now for the text area

File: src/api/app/views/webui/user/_basic_info.html.haml [hunk 2]
```diff
@@ -3,12 +3,22 @@
...
+    - if User.possibly_nobody.is_admin?
+      = link_to('#', title: 'Delete user', class: 'ms-2', data: { 'bs-toggle': 'modal', 'bs-target': '#delete-user-modal' }) do
+        %i.fas.fa-times-circle.text-danger
+        %span.nav-item-name.text-danger Delete
+      = render DeleteConfirmationDialogComponent.new(modal_id: 'delete-user-modal', method: :delete,
+                                                     options: { modal_title: 'Delete user?', remote: false,
+                                                                confirmation_text: "Please confirm deletion of '#{user.login}'",
+                                                                action: user_path(user) }) do |component|
+        - component.with_text_area do
+          = label_tag('Adminnote', 'Adminnote:', for: 'adminnote')
```
Comment (eduardoj) [inline src/api/app/views/webui/user/_basic_info.html.haml:16]:
```suggestion
          = label_tag('Adminnote', 'Admin note:', for: 'adminnote')
```
I don't think _Adminnote_ exists as a proper English word...

File: src/api/app/views/webui/users/index.html.haml [hunk 3]
```diff
@@ -27,7 +27,10 @@
     = render DeleteConfirmationDialogComponent.new(modal_id: 'delete-user-modal',
                                                    method: :delete,
                                                    options: { modal_title: 'Delete user?', remote: false,
-                                                              confirmation_text: sanitize(confirmation_text, tags: ['span']) })
+                                                              confirmation_text: sanitize(confirmation_text, tags: ['span']) }) do |component|
+      - component.with_text_area do
+        = label_tag('Adminnote', 'Adminnote:', for: 'adminnote')
```
Comment (eduardoj) [inline src/api/app/views/webui/users/index.html.haml:32]:
```suggestion
        = label_tag('Adminnote', 'Admin note:', for: 'adminnote')
```
Also here.

File: src/api/app/views/webui/user/_basic_info.html.haml (hunk 2 above)
Comment (krauselukas) [reply src/api/app/views/webui/user/_basic_info.html.haml]: @eduardoj done

**Pull request merged**